### PR TITLE
Concurrent X11 sessions fix

### LINKED ISSUE #10457: Can't have multiple X sessions via teleport
If someone else is using :10    connections fails.

X11 forwarding request failed on channel 0
debug2: channel 0: rcvd close
debug2: channel 0: output open -> drain
debug2: chan_shutdown_read: channel 0: (i0 o1 sock -1 wfd 3 efd 7 [write])
debug2: channel 0: input open -> closed
debug2: channel 0: obuf_empty delayed efd 7/(112)
listen unix /tmp/.X11-unix/X10: bind: address already in use, hostname can't be empty for an XServer tcp socket
                                                                                                               debug2: channel 0: written 112 to efd 7
debug2: channel 0: obuf empty
debug2: chan_shutdown_write: channel 0: (i3 o1 sock -1 wfd 6 efd 7 [write])
debug2: channel 0: output drain -> closed
debug2: channel 0: almost dead
debug2: channel 0: gc: notify user
debug2: channel 0: gc: user detached
debug2: channel 0: send close
debug2: channel 0: is dead
debug2: channel 0: garbage collecting
debug1: channel 0: free: client-session, nchannels 1

### PR DESCRIPTION
**Pull request by Joerger**
OpenXServerListener would mistakenly stop searching for open XServer sockets after encountering the first "bind: address already in use" error.

Closes https://github.com/gravitational/teleport/issues/10457

### CODE REVIEW

File: lib/sshutils/x11/display.go [hunk 1]
```diff
@@ -86,13 +88,17 @@ func (d *Display) Dial() (XServerConn, error) {
...
+// Listen opens an XServer listener. It will attempt to listen on the display
+// address for both tcp and unix and return an aggregate error, unless one
+// results in an addr in use error.
 func (d *Display) Listen() (XServerListener, error) {
 	unixSock, unixErr := d.unixSocket()
 	if unixErr == nil {
 		var l *net.UnixListener
 		if l, unixErr = net.ListenUnix("unix", unixSock); unixErr == nil {
 			return &xserverUnixListener{l}, nil
+		} else if errors.Is(unixErr, syscall.EADDRINUSE) {
```
Comment (r0mant) [inline lib/sshutils/x11/display.go:100]: @Joerger I'm not sure I fully understand how this fixes the issue with concurrent sessions tbh. Previously, we would still return any error returned by Listen* call below as a part of the aggregate error. With this fix seems like we just return an error sooner. I'm probably missing something, could you explain?

Comment (Joerger) [reply lib/sshutils/x11/display.go:100]: The function that calls `display.Listen` is `x11.OpenXServerListener`, which expects `EADDRINUSE` errors for x11 displays which are already being used. However, `display.Listen` was returning an aggregate error which does not resolve to `syscall.EADDRINUSE`. This change fixes that by returning the `syscall.EADDRINUSE` error immediately, not in aggregate.

[```
for displayNumber := displayOffset; displayNumber <= maxDisplay; displayNumber++ {
	display := Display{DisplayNumber: displayNumber}
	if l, err := display.Listen(); err == nil {
		return l, display, nil
	} else if !errors.Is(err, syscall.EADDRINUSE) {
		return nil, Display{}, trace.Wrap(err)
	}
}
```](https://github.com/gravitational/teleport/blob/ea810d30d99f26e58a190edc5facfbe0c09ea5e5/lib/sshutils/x11/conn.go#L81-L88)

Comment (r0mant) [reply lib/sshutils/x11/display.go:100]: Okay, makes sense.

**Review by r0mant — APPROVED** [review]

**Review by russjones — APPROVED** [review]

**Comment by russjones** [comment]
@Joerger Let's backport this to `branch/v8` next week.

**Pull request merged**